internal/services/keyvault/key_vault_managed_storage_account.go

```diff
@@ -169,12 +169,8 @@ func resourceKeyVaultManagedStorageAccountCreateUpdate(d *pluginsdk.ResourceData
 		return fmt.Errorf("cannot read Managed Storage Account ID %q (Key Vault %q)", name, *keyVaultId)
 	}
 
-	storageId, err := commonids.ParseStorageAccountID(*read.ID)
-	if err != nil {
-		return err
-	}
-
-	d.SetId(storageId.ID())
+	// This need to be set as this id instead of storage account resource id
+	d.SetId(*read.ID)
```

Comment:
rather than doing this, we can update the ID parser to be: `keyVaultParse.ParseOptionallyVersionedNestedItemID` and then call `.ID()` on this - can we update this?

Reply:
Ok, I will try to update this later today.

Reply:
Hi @tombuildsstuff, after switching to the new parser, following errors will be generated.
```
=== RUN   TestAccKeyVaultManagedStorageAccountSasTokenDefinition_basic
=== PAUSE TestAccKeyVaultManagedStorageAccountSasTokenDefinition_basic
=== CONT  TestAccKeyVaultManagedStorageAccountSasTokenDefinition_basic
    testcase.go:113: Step 1/2 error: Error running apply: exit status 1
        
        Error: key vault nested item should contain 2 or 3 segments, found 4 segment(s) in "https://acctestkv-hokax.vault.azure.net/storage/acctestKVstorage/sas/acctestKVsasdefinition"
        
          with azurerm_key_vault_managed_storage_account_sas_token_definition.test,
          on terraform_plugin_test.tf line 111, in resource "azurerm_key_vault_managed_storage_account_sas_token_definition" "test":
         111: resource "azurerm_key_vault_managed_storage_account_sas_token_definition" "test" {
        
--- FAIL: TestAccKeyVaultManagedStorageAccountSasTokenDefinition_basic (298.90s)

FAIL
```

Reply:
@liuwuliuyun `azurerm_key_vault_managed_storage_account_sas_token_definition` has [its own parser `SasDefinitionID`](https://github.com/hashicorp/terraform-provider-azurerm/blob/f8b2767b5f4e78ee0f6dbc122bb8acbcf5fceed7/internal/services/keyvault/parse/sas_definition.go#L12).
That parser is being used in `resource...Read`, `resource...Delete` and the importer, but not `resource...CreateUpdate`.

*Just hit this issue while creating `azurerm_key_vault_managed_storage_account` with terraform and found this PR is just the fix I need.*

Reply:
@bianyifan Thanks. If you are in a rush, you could try https://registry.terraform.io/providers/Azure/azapi/latest/docs to build this resource. I will look into this id and try to fix this issue.

Reply:
@liuwuliuyun You're welcome. I've worked around this issue with `terraform import`.